### PR TITLE
EVG-18870 Preserve directory structure of files uploaded to storybook

### PR DESCRIPTION
**Pull request by khelif96**
EVG-18870

### Description 
Utilize the changes introduced by @malikchaya2  in EVG-17334 to upload files to storybook

### CODE REVIEW

File: .evergreen.yml [hunk 1]
```diff
@@ -66,57 +66,82 @@ functions:
...
+        ["parsley/storybook-static/*.html"]
       remote_file: parsley/${task_id}/storybook/
       bucket: mciuploads
       content_type: text/html
       permissions: public-read
       display_name: storybook_
+      preserve_path: true
 
-  # Ideally we can get rid of this command upon the completion of EVG-17334.
   attach-storybook-assets-js:
```
Comment (sophstad) [inline .evergreen.yml:77]: What do you think of combining all the `attach-storybook-` functions into one function with multiple commands? I did this in https://github.com/evergreen-ci/spruce/pull/1640 and I think it is a little more readable.

File: .storybook/main.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // eslint-disable-next-line import/no-import-module-exports
 import type { StorybookConfig } from "@storybook/react-vite";
+import { defineConfig, mergeConfig } from "vite";
```
Comment (sophstad) [inline .storybook/main.ts:3]: Looks like this may have been accidentally included

File: .evergreen.yml (hunk 1 above)
Comment (khelif96) [reply .evergreen.yml]: Good call done

File: .evergreen.yml [hunk 3]
```diff
@@ -60,63 +60,80 @@ functions:
...
+        local_files_include_filter:
+          ["parsley/storybook-static/**/*.json"]
+        remote_file: parsley/${task_id}/storybook/
+        bucket: mciuploads
+        content_type: application/json
+        permissions: public-read
+        display_name: storybook_json_
+        preserve_path: true
+
+  link-to-storybook:
```
Comment (sophstad) [inline .evergreen.yml:129]: very nice 😎

**Review by sophstad — APPROVED** [review]

**Pull request merged**